### ISSUE TITLE
wrong content-type reponse header

### ISSUE THREAD

**Issue by btrem** [11ty/eleventy#1599]
**Describe the bug**
eleventy server sends the wrong response header for web pages whose permalink is "extensionless" e.g., /foo and not /foo.html

**To Reproduce**
1. create foo.md; set permalink to /foo
2. run 11ty server `npx @11ty/eleventy --serve`
3. open page in browser, e.g. localhost:xxxx/foo

**Expected behavior**
View web page in browser.

**Actual behavior**
Dialog asking if I want to save the file.

**Environment:**
- OS and Version: Mac OSX 10.13.6 'High Sierra'
- Eleventy Version 0.11.1

**Additional context**
The content-type response header for the request as seen by Firefox Developer Edition:
`Content-Type: application/octet-stream`

**Comment by varsubham** [comment]
@btrem I think it is clearly mentioned in the docs that if you leave off the trailing slash, than the browser may attempt to download the file instead of displaying it.
Please check it out [here](https://www.11ty.dev/docs/permalinks/#remapping-output-(permalink))

![Capture96](https://user-images.githubusercontent.com/47079909/105580617-7184f480-5db3-11eb-96bd-3dc708fcb33f.PNG)

**Comment by btrem** [comment]
Right. I did see that, but didn't connect it to my issue. I assumed that was in case you wanted `/foo` to be equivalent to `/foo/`. But I don't want them to be equivalent. In my case, `/foo/` does not exist on my server; but I want `/foo` to be my page. I don't think this is an edge case. Lots of sites use "extensionless" urls, ones that don't end in extraneous characters like "/" or ".html".

On my server, I *have* "done extra work" to send the correct Content-Type header. (Really, it's not much more work. Just change the default type, and done.) So can I replicate that with 11ty --serve? I think it's wrong for 11ty --serve to send a  `Content-Type: application/octet-stream` for a web page.

**Comment by paulshryock** [comment]
@btrem, have you tried just setting the permalink to `/foo/` with a trailing slash, and then visiting your page in the browser at the desired extension-less location, `/foo` (without adding the trailing slash)? It should just work.

**Comment by btrem** [comment]
Setting the permalink to `/foo/` causes 11ty to write the page /foo/index.html. Servers might be configured to serve index.html for a request that resolves to a directory. Servers might also be configured to handle missing files like this: if there's a request for `/foo`; and there's no file named foo; and there is a *directory* named foo; then the server redirects the request to `/foo/`. And since the new request resolves to a directory, per the directory rule, the default file index.html is served.

I assume this is what you mean by "It should just work." But this does not work. My urls do **not** end in a slash. The urls are `/foo`, `/bar`, etc. I do not want to redirect my urls to `/foo/` and `/bar/`. I want my urls to remain as they are. I can keep them as they are, but 11ty sends the wrong Content-Type header, which means that 11ty --serve is broken for my url setup.

**Comment by paulshryock** [comment]
Gotcha. [Eleventy uses BrowserSync](https://www.11ty.dev/docs/usage/#re-run-eleventy-when-you-save) under the hood, so if anything's broken, I think it's BrowserSync.

You can override [BrowserSync options](https://browsersync.io/docs/options) via [Eleventy config](https://www.11ty.dev/docs/watch-serve/#override-browsersync-server-options). But I haven't found a way to serve custom response headers in BrowserSync. I don't think this is something that Eleventy can fix, as long as it continues to use BrowserSync as a server (though I could be wrong about that). If you find a way to serve custom response headers with BrowserSync, please let me know.

So instead of using `eleventy --serve`, I usually set up a local server like [connect](https://github.com/senchalabs/connect) or [Express](https://expressjs.com/) where I have a bit more control over things like response headers.

**Comment by btrem** [comment]
Ok, that looks promising. I'll take a look at that and report back here with any results.

**Comment by btrem** [comment]
It needed a call to `next()` (otherwise, `11ty --serve` just hangs). I also added a regex test to the req. Here's what I'm using:

    eleventyConfig.setBrowserSyncConfig({
        middleware: [
            function (req, res, next) {
                if (/^[^.]+$/.test(req.url)) {
                    res.setHeader('Content-Type', 'text/html; charset=utf-8');
                }
                next();
            }
        ]
    });
Now /foo returns a text/html page that I can view in my browser. I wrote it up in an [article on my website](https://btrem.com/articles/2021/02-content-type-11ty). The beginning (how to configure Apache) might be review for GitHub users. So feel free to skip to the end. Or just use the code posted in this reply.
